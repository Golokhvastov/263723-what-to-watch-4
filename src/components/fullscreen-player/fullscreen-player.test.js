import React from "react";
import renderer from "react-test-renderer";
import FullscreenPlayer from "./fullscreen-player.jsx";

const movie = {
  title: `Test1`,
  previewImage: `test2.jpg`,
  preview: `test3.jpg`,
  src: `test4.jpg`,
  runTime: 55,
};

describe(`FullscreenPlayer render correctly`, () => {
  it(`progress = 0, isLoading = false, isPlaying = true`, () => {
    const tree = renderer
      .create(
          <FullscreenPlayer
            movie = {movie}
            onExitClick = {() => {}}
            progress = {0}
            isLoading = {false}
            isPlaying = {true}
            onPlayClick = {() => {}}
            onPauseClick = {() => {}}
          >
            <video />
          </FullscreenPlayer>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`progress = 11, isLoading = false, isPlaying = false`, () => {
    const tree = renderer
      .create(
          <FullscreenPlayer
            movie = {movie}
            onExitClick = {() => {}}
            progress = {11}
            isLoading = {false}
            isPlaying = {false}
            onPlayClick = {() => {}}
            onPauseClick = {() => {}}
          >
            <video />
          </FullscreenPlayer>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`progress = 22, isLoading = true, isPlaying = true`, () => {
    const tree = renderer
      .create(
          <FullscreenPlayer
            movie = {movie}
            onExitClick = {() => {}}
            progress = {22}
            isLoading = {true}
            isPlaying = {true}
            onPlayClick = {() => {}}
            onPauseClick = {() => {}}
          >
            <video />
          </FullscreenPlayer>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`progress = 100, isLoading = true, isPlaying = false`, () => {
    const tree = renderer
      .create(
          <FullscreenPlayer
            movie = {movie}
            onExitClick = {() => {}}
            progress = {100}
            isLoading = {true}
            isPlaying = {false}
            onPlayClick = {() => {}}
            onPauseClick = {() => {}}
          >
            <video />
          </FullscreenPlayer>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
