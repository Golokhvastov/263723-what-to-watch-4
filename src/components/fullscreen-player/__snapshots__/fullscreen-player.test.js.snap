// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullscreenPlayer render correctly progress = 0, isLoading = false, isPlaying = true 1`] = `
<div
  className="player"
>
  <video />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max={3300}
          value={0}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        00:55:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="21"
          viewBox="0 0 14 21"
          width="14"
        >
          <use
            xlinkHref="#pause"
          />
        </svg>
        <span>
          Pause
        </span>
      </button>
      <div
        className="player__name"
      >
        Test1
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`FullscreenPlayer render correctly progress = 11, isLoading = false, isPlaying = false 1`] = `
<div
  className="player"
>
  <video />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max={3300}
          value={11}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0.33333333333333337%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        00:54:49
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Test1
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`FullscreenPlayer render correctly progress = 22, isLoading = true, isPlaying = true 1`] = `
<div
  className="player"
>
  <video />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max={3300}
          value={22}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0.6666666666666667%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        00:54:38
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="21"
          viewBox="0 0 14 21"
          width="14"
        >
          <use
            xlinkHref="#pause"
          />
        </svg>
        <span>
          Pause
        </span>
      </button>
      <div
        className="player__name"
      >
        Test1
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`FullscreenPlayer render correctly progress = 100, isLoading = true, isPlaying = false 1`] = `
<div
  className="player"
>
  <video />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max={3300}
          value={100}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "3.0303030303030303%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        00:53:20
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Test1
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
